rasterio/merge.py

```diff
@@ -45,6 +47,10 @@ def merge(datasets, bounds=None, res=None, nodata=None, precision=7, indexes=Non
         in the first input raster.
     indexes : list of ints or a single int, optional
         bands to read and merge
+    method : str or callable
+        pre-defined or custom merging method
+        callable must have signature
+        (old_data, new_data, old_nodata, new_nodata)
```

Comment:
@j08lue I think we need to do a nested docstring here, something like

```suggestion

        function(old_data, new_data, old_nodata, new_nodata)
        
        Parameters
        -----------
        old_data : array_like
        ...
```

Reply:
Information about the expected types in the callback would be key.

Reply:
done